lib/ansible/config/manager.py

```diff
@@ -547,8 +547,18 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                     invalid_choices = value not in defs[config]['choices']
 
                 if invalid_choices:
+
+                    if isinstance(defs[config]['choices'], Mapping):
+                        valid = ', '.join(defs[config]['choices'].keys())
+                    elif isinstance(defs[config]['choices'], string_types):
+                        valid = defs[config]['choices']
+                    elif isinstance(defs[config]['choices'], Sequence):
+                        valid = ', '.join(defs[config]['choices'])
```

Comment:
Since choices can be int/bool/floats, using to_text before joining would fix TypeError for those:
```python
valid = ', '.join(to_text(choice) for choice in defs[config]['choices'])
```

Reply:
good catch